pkg/auth/roundtripper.go

```diff
@@ -0,0 +1,46 @@
+package auth
+
+import (
+	"net/http"
+
+	"github.com/go-kit/kit/log"
+)
+
+type InstantQueryRoundTripper struct {
```

Comment:
Naming nit: this no longer has anything to do with instant queries in particular. Maybe we can rename this to `BearerTokenRoundTripper` for example

Reply:
Otherwise perfect, this abstraction is exactly what we needed